diff --git a/cmds/bootanimation/BootAnimation.cpp b/cmds/bootanimation/BootAnimation.cpp
index bf1863e83d43..352841c0415d 100644
--- a/cmds/bootanimation/BootAnimation.cpp
+++ b/cmds/bootanimation/BootAnimation.cpp
@@ -76,6 +76,7 @@ static const char OEM_BOOTANIMATION_FILE[] = "/oem/media/bootanimation.zip";
 static const char PRODUCT_BOOTANIMATION_DIR[] = "/product/media/";
 static const char PRODUCT_BOOTANIMATION_DARK_FILE[] = "bootanimation-dark.zip";
 static const char PRODUCT_BOOTANIMATION_FILE[] = "bootanimation.zip";
+static const char SYSTEM_BOOTANIMATION_DARK_FILE[] = "/system/media/bootanimation-dark.zip";
 static const char SYSTEM_BOOTANIMATION_FILE[] = "/system/media/bootanimation.zip";
 static const char APEX_BOOTANIMATION_FILE[] = "/apex/com.android.bootanimation/etc/bootanimation.zip";
 static const char OEM_SHUTDOWNANIMATION_FILE[] = "/oem/media/shutdownanimation.zip";
@@ -683,9 +684,11 @@ void BootAnimation::findBootAnimationFile() {
     const std::string productBootanimationFile = PRODUCT_BOOTANIMATION_DIR +
         android::base::GetProperty("ro.product.bootanim.file", playDarkAnim ?
         PRODUCT_BOOTANIMATION_DARK_FILE : PRODUCT_BOOTANIMATION_FILE);
-    static const std::vector<std::string> bootFiles = {
+    const char* systemBootanimationFile = playDarkAnim ?
+        SYSTEM_BOOTANIMATION_DARK_FILE : SYSTEM_BOOTANIMATION_FILE;
+    const std::vector<std::string> bootFiles = {
         APEX_BOOTANIMATION_FILE, productBootanimationFile,
-        OEM_BOOTANIMATION_FILE, SYSTEM_BOOTANIMATION_FILE
+        OEM_BOOTANIMATION_FILE, systemBootanimationFile
     };
     static const std::vector<std::string> shutdownFiles = {
         PRODUCT_SHUTDOWNANIMATION_FILE, OEM_SHUTDOWNANIMATION_FILE, SYSTEM_SHUTDOWNANIMATION_FILE, ""
diff --git a/packages/SystemUI/compose/features/src/com/android/systemui/qs/ui/composable/QuickSettingsScene.kt b/packages/SystemUI/compose/features/src/com/android/systemui/qs/ui/composable/QuickSettingsScene.kt
index 38855b2bae43..2b9bcc301b04 100644
--- a/packages/SystemUI/compose/features/src/com/android/systemui/qs/ui/composable/QuickSettingsScene.kt
+++ b/packages/SystemUI/compose/features/src/com/android/systemui/qs/ui/composable/QuickSettingsScene.kt
@@ -225,8 +225,7 @@ private fun ContentScope.QuickSettingsScene(
     val animatedBlurRadiusPx: Float by
         animateFloatAsState(targetValue = targetBlur, label = "QS-blurRadius")
     val onlyPunchHolesInThisScene =
-        layoutState.isTransitioningBetween(Scenes.Gone, Scenes.QuickSettings) ||
-            layoutState.isTransitioningBetween(Scenes.Lockscreen, Scenes.QuickSettings)
+        layoutState.isTransitioningBetween(Scenes.Lockscreen, Scenes.QuickSettings)
     Box(
         modifier
             .fillMaxSize()
@@ -343,7 +342,7 @@ private fun ContentScope.QuickSettingsScene(
                 viewModel = notificationsPlaceholderViewModel,
                 notificationRulesParentViewModel = notificationRulesParentViewModel,
                 jankMonitor = jankMonitor,
-                shouldPunchHoleBehindScrim = true,
+                shouldPunchHoleBehindScrim = onlyPunchHolesInThisScene,
                 shouldFillMaxHeight = true,
                 isTransparencyEnabled = viewModel.isTransparencyEnabled,
                 stackTopPadding = dimensionResource(id = R.dimen.notification_side_paddings_single),
diff --git a/packages/SystemUI/compose/features/src/com/android/systemui/shade/ui/composable/ShadeScene.kt b/packages/SystemUI/compose/features/src/com/android/systemui/shade/ui/composable/ShadeScene.kt
index 0a42c6f51eb0..b32a1bbebad1 100644
--- a/packages/SystemUI/compose/features/src/com/android/systemui/shade/ui/composable/ShadeScene.kt
+++ b/packages/SystemUI/compose/features/src/com/android/systemui/shade/ui/composable/ShadeScene.kt
@@ -313,8 +313,7 @@ private fun ContentScope.SingleShade(
     LaunchedEffectWithLifecycle(Unit) { viewModel.detectShadeModeChanges() }
 
     val onlyPunchHolesInThisScene =
-        layoutState.isTransitioningBetween(Scenes.Gone, Scenes.Shade) ||
-            layoutState.isTransitioningBetween(Scenes.Lockscreen, Scenes.Shade)
+        layoutState.isTransitioningBetween(Scenes.Lockscreen, Scenes.Shade)
     val mediaInRow = viewModel.showMediaInRow
     val notificationStackPadding = dimensionResource(id = R.dimen.notification_side_paddings_single)
 
@@ -502,7 +501,7 @@ private fun ContentScope.SingleShade(
                     stackScrollView = notificationStackScrollView,
                     viewModel = notificationsPlaceholderViewModel,
                     notificationRulesParentViewModel = notificationRulesParentViewModel,
-                    shouldPunchHoleBehindScrim = true,
+                    shouldPunchHoleBehindScrim = onlyPunchHolesInThisScene,
                     shouldContentFillMaxSize = true,
                     shouldScrimBackgroundFillMaxHeight = true,
                     isTransparencyEnabled = viewModel.isTransparencyEnabled,
